fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -290,6 +327,24 @@ public DataApiRequest(
     }
 
     /**
+     * Times the execution of the specified producer, making sure to stop timing in a finally clause.
+     *
+     * @param generator  The producer to be timed
+     * @param timerName  The name to use for the timer
+     * @param <T>  The type of the object produced
+     *
+     * @return The value returned by the generator
+     */
+    private <T> T timeGenerator(Producer<T> generator, String timerName) {
```

Comment:
While this solution is cute, it uglies up the calls everywhere, making the code much harder to understand. Instead, if we want to do this, we should use annotations on the "generator" methods. Take a look at the `@Timed` annotation in the Metrics library to see how they are doing it. Having a similar annotation for the `RequestLog` may be exactly what we're looking for.